api/src/main/java/org/phenotips/variantStoreIntegration/DefaultVCFUploadManager.java

```diff
@@ -96,29 +106,44 @@ public void initialize() throws InitializationException
     /**
      * {@inheritDoc}
      *
-     * @see org.phenotips.variantStoreIntegration.VCFUploadManager#uploadVCF(org.phenotips.data.Patient, java.io.File)
      */
     @Override
-    public void uploadVCF(Patient patient, Path filePath)
+    public void uploadVCF(String patientID, String filePath) throws Exception
     {
-        String id = patient.getId();
+        Patient patient = this.pr.getPatientById(patientID);
 
-        if (this.currentUploads.get(id) != null) {
-            this.logger.warn("Tried to upload VCF of {} while it was already uploading", patient.toString());
-            return;
-        } else if (this.currentRemovals.get(id) != null) {
-            this.logger.warn("Tried to upload VCF of {} while it was being removed", patient.toString());
-            return;
+        if (patient == null){
+            this.logger.warn("No patient found with the id: {}", patientID);
+            throw new Exception("Could not find the patient with ID: " + patientID);
         }
 
-        XWikiContext context = (XWikiContext) this.execution.getContext().getProperty("xwikicontext");
+        File vcfFile = new File(filePath);
+        if (!vcfFile.exists() || !vcfFile.isFile()) {
+            this.logger.warn("Attempted to upload an invalid VCF file");
+            throw new Exception("No file was found at: " + filePath);
+        }
 
-        boolean isPublic = DefaultVCFUploadManager.resolvePatientPermission(id);
+        if (this.currentUploads.get(patientID) != null) {
+            this.logger.warn("Tried to upload VCF of {} while it was already uploading", patientID);
+            return;
+        } else if (this.currentRemovals.get(patientID) != null) {
+            this.logger.warn("Tried to upload VCF of {} while it was being removed", patientID);
```

Comment:
Same here. Maybe these should be queued, so a new job should start only after the current job finishes.